### PR TITLE
feat: add coordinator wait/sleep MCP tool (#371)

### LINKED ISSUE #371: feat: add coordinator wait/sleep MCP tool for long-running workflow monitoring
## Problem

The coordinator agent orchestrates long-running workflows (extraction runs, benchmarks) that take 2-4 hours. Currently, monitoring requires repeatedly dispatching subagents to check status, which:
- Burns tokens on redundant status checks
- Subagents timeout (~10-15 min execution limit) before long waits complete
- The coordinator has no way to pause and resume

## Proposed Solution

Create a lightweight MCP server exposing a `wait` tool:

### Tool: `wait`
- **seconds** (int, required): Duration to sleep (1-14400, i.e. max 4 hours)
- **message** (string, optional): Log message describing what we are waiting for
- Returns: Confirmation with actual elapsed time

### Implementation
- Single Python file: `tools/mcp/wait_server.py`
- Uses `asyncio.sleep()` for non-blocking wait
- Safety cap at 4 hours max
- Graceful cancellation support
- Register in `.vscode/mcp.json`

### Coordinator Update
Update coordinator mode instructions to include scheduling responsibility:
- Use `wait` tool instead of repeated subagent dispatch for long-running process monitoring
- Pattern: estimate remaining time, wait 80% of estimate, then dispatch status check

## Acceptance Criteria

- [ ] MCP server starts and registers correctly in VS Code
- [ ] `wait` tool accepts seconds (1-14400) and optional message
- [ ] Returns actual elapsed time on completion
- [ ] Rejects invalid inputs (negative, > 14400, non-integer)
- [ ] Coordinator instructions updated with scheduling pattern
- [ ] Works with existing MCP configuration

### PR DESCRIPTION
**Pull request by daviburg**
## Summary

Adds a lightweight MCP server exposing a `wait` tool that lets the coordinator agent pause for a specified duration before resuming work. This enables token-efficient monitoring of long-running processes (extraction runs, benchmarks) that take 2-4 hours.

### Changes
- `tools/mcp/wait_server.py` — MCP server with `wait(seconds, message)` tool
- `.vscode/mcp.json` — MCP server registration
- `requirements-mcp.txt` — MCP package dependency
- `tests/test_wait_server.py` — Unit tests for input validation and timing (6 tests)
- `docs/architecture.md` — Added MCP Tools section and tool script entry
- `docs/usage.md` — Added MCP Tools section with setup and usage docs
- `.github/copilot-instructions.md` — Added `requirements-mcp.txt` and `.vscode/mcp.json` to file conventions table

### How it works
The coordinator calls `wait(seconds=3600, message="extraction run")`, which sleeps for the specified duration using `asyncio.sleep()`, then returns confirmation with actual elapsed time. Safety cap at 4 hours (14400 seconds). Invalid inputs (negative, zero, over max) return error messages.

### Coordinator scheduling pattern
1. Launch a detached extraction run
2. Estimate remaining time (e.g. 2 hours)
3. Call `wait(seconds=5760, message="extraction run ~80% of estimated 2h")`
4. After wait completes, dispatch a status check subagent

Closes #371

### CODE REVIEW

**Review by daviburg — COMMENTED** [review]
**[@reviewer]** Review verdict: REQUEST CHANGES

## Blocking Issues

1. **Unresolved automated review comments** — 3 Copilot comments with no replies. All PR comment threads must have replies before merge.
2. **Missing acceptance criterion** — coordinator.agent.md not updated with scheduling pattern per issue #371.

## Suggestions (non-blocking)

- .vscode/mcp.json uses bare python — consider venv path
- roadmap.md not updated with MCP tooling capability
- Boolean validation: type(seconds) is int preferred over isinstance
- CancelledError handling for clean cancellation logging

## Positive

- Code minimal and well-structured, security cap enforced, tests cover edge cases, docs updated (architecture, usage, copilot-instructions).

Fix the two blocking items and this is ready to approve.

File: tools/mcp/wait_server.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    benchmarks) instead of repeatedly dispatching subagents.
+
+    Args:
+        seconds: Duration to wait (1-14400, max 4 hours)
+        message: Optional description of what you're waiting for
+
+    Returns:
+        Confirmation with actual elapsed time
+    """
+    if not isinstance(seconds, int) or seconds < 1:
```
[the remark this answers is not in the document]
Comment (daviburg) [reply tools/mcp/wait_server.py]: **[@developer]** Fixed in 56da599: Added `isinstance(seconds, bool)` guard before the `isinstance(seconds, int)` check. Also added `test_rejects_bool` covering both `True` and `False`.

File: tools/mcp/wait_server.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    if message:
+        print(f"[wait-server] Waiting {seconds}s: {message}")
+
+    await asyncio.sleep(seconds)
+
+    elapsed = time.monotonic() - start
+    result = f"Wait complete. Requested: {seconds}s, actual elapsed: {elapsed:.1f}s"
+    if message:
+        result += f" (was waiting for: {message})"
+    return result
```
[the remark this answers is not in the document]
Comment (daviburg) [reply tools/mcp/wait_server.py]: **[@developer]** Fixed in 56da599: Added `asyncio.CancelledError` handling around the `await asyncio.sleep(seconds)` call. On cancellation, returns elapsed time and the original request. Added `test_wait_cancelled` to verify.

File: tests/test_wait_server.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    """Test the wait tool function directly (without MCP transport)."""
+
+    def test_short_wait(self):
+        """A 1-second wait should complete in ~1 second."""
+        start = time.monotonic()
+        result = asyncio.run(wait(1))
+        elapsed = time.monotonic() - start
+        assert "Wait complete" in result
+        assert elapsed >= 0.9
+        assert elapsed < 2.0
```
[the remark this answers is not in the document]
Comment (daviburg) [reply tests/test_wait_server.py:27]: **[@developer]** Acknowledged. The 1-second sleep with 0.9-2.0 window is intentionally generous for CI variability. The tool is inherently time-based (its entire purpose is sleeping), so a real-sleep smoke test is appropriate here. Mocking `asyncio.sleep` would test nothing meaningful — we need to verify the actual wait behavior. The 2x upper bound accommodates even heavily loaded CI runners.

**Review by daviburg — COMMENTED** [review]
**[@reviewer]** Re-review APPROVED. All prior blockers resolved: (1) bool validation guard with isinstance(seconds, bool) check, (2) asyncio.CancelledError handled gracefully, (3) all 3 Copilot comment threads have replies, (4) coordinator.agent.md has Scheduling/Long Waits section, (5) docs/architecture.md and docs/usage.md updated per Rule 8, (6) tests cover bool rejection, boundary, and cancellation, (7) CI green (CodeQL + Quality Gate). Ready to merge.
